ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configurator.ts

```diff
@@ -0,0 +1,34 @@
+import {Configurator} from './configurator';
+import {CustomYasguiConfiguration} from '../../../models/custom-yasgui-configuration';
+
+/**
+ * Manages all top configuration of yasgui.
+ *
+ */
+class YasguiConfiguratorDefinition implements Configurator {
+
+  private defaultYasguiConfig = {
+    copyEndpointOnNewTab: true
+  }
+
+  private defaultRequestConfig = {
+    endpoint: '',
+    method: 'POST',
+    headers: () => {
+      return {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'Accept': 'application/sparql-results+json',
+        'X-GraphDB-Local-Consistency': 'updating'
+      };
+    }
+  }
+
+  config(yasguiConfig, customYasguiConfig: CustomYasguiConfiguration) {
+    const result = {... yasguiConfig, ... this.defaultYasguiConfig};
+    result.requestConfig = {... this.defaultRequestConfig}
+    result.requestConfig.endpoint = customYasguiConfig.endpoint;
```

Comment:
Why don't just extend the request config with the provided one. SHould we add here each and every possible parameter one by one?

Reply:
done